tests/pkg-layering/main.yml

```diff
@@ -0,0 +1,319 @@
+        installed_json: "{{ installed.stdout | from_json }}"
+
+    - name: Set packages when deployment 0 is booted
+      set_fact:
+        installed_pkgs: "{{ installed_json['deployments'][0]['packages'] }}"
+      when: installed_json['deployments'][0] is defined and installed_json['deployments'][0]['booted']
+
+    - name: Set packages when deployment 1 is booted
+      set_fact:
+        installed_pkgs: "{{ installed_json['deployments'][1]['packages'] }}"
+      when: installed_json['deployments'][1] is defined and installed_json['deployments'][1]['booted']
+
+    - name: Fail if {{ g_pkg1 }} is in rpm-ostree status output
+      fail:
+        msg: "{{ g_pkg1 }} in rpm-ostree status output"
+      when: "'{{ g_pkg1 }}' in installed_pkgs"
+
+    - name: Fail if {{ g_pkg1 }} is installed
+      command: command -v {{ g_pkg1 }}
+      register: binary
+      failed_when: binary.rc != 1
+    
+    # Verify a dry-run of rpm-ostree install|uninstall does not affect an existing package that has been layered, but not rebooted.
+    - name: Test Install {{ g_pkg1 }} and do not reboot
+      command: rpm-ostree install {{ g_pkg1 }}
+
+    - name: Dry run install {{ g_pkg2 }}
+      command: rpm-ostree install {{ g_pkg2 }} --dry-run
+      register: dryrun
+
+    - name: Fail if Exiting because of is not in output
+      fail:
+        msg: "Dry run output does not contain 'Exiting because'"
+      when: "'Exiting because' not in dryrun.stdout"
+
+    - name: Get rpm-ostree status --json output
+      command: rpm-ostree status --json
+      register: installed
+  
+    - name: Convert rpm-ostree status output to jinja2 json
+      set_fact:
+        pending_json: "{{ installed.stdout | from_json }}"
+
+    - name: Set packages when deployment 0 is not booted
+      set_fact:
+        pending_pkgs: "{{ pending_json['deployments'][0]['packages'] }}"
+      when: pending_json['deployments'][0] is defined and not pending_json['deployments'][0]['booted']
+
+    - name: Set packages when deployment 1 is not booted
+      set_fact:
+        pending_pkgs: "{{ pending_json['deployments'][1]['packages'] }}"
+      when: pending_json['deployments'][1] is defined and not pending_json['deployments'][1]['booted']
+
+    - name: Fail if {{ g_pkg2 }} is in rpm-ostree status output for pending packages
+      fail:
+        msg: "{{ g_pkg2 }} in rpm-ostree status output"
+      when: "'{{ g_pkg2 }}' in pending_pkgs"
+
+    - name: Fail if {{ g_pkg1 }} is not in rpm-ostree status output for pending packages
+      fail:
+        msg: "{{ g_pkg1 }} not in rpm-ostree status output"
+      when: "'{{ g_pkg1 }}' not in pending_pkgs"
+
+    - name: Fail if {{ g_pkg1 }} is installed
+      command: command -v {{ g_pkg1 }}
+      register: binary
+      failed_when: binary.rc != 1
+
+    # Verify that after additional repos are enabled (i.e. EPEL), rpm-ostree install|uninstall can be successfully used for packages in those repos.
+    - name: Retrieve EPEL rpm
```

Comment:
In somewhere else, I forget now, I just wrote the `.repo` and gpg key via `copy/contents` rather than fetching the RPM.  There's also various ansible modules for writing yum repo configs that would make sense.